Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt

```diff
@@ -130,11 +136,106 @@ fun IncludeCertificateQrcodeCardBinding.bindValidityViews(
             startValidationCheckButton.isVisible = isPersonDetails
         }
     }
+}
+
+@Suppress("LongParameterList", "ComplexMethod")
+fun PersonOverviewItemBinding.setUIState(
+    primaryCertificate: CwaCovidCertificate,
+    secondaryCertificate: CwaCovidCertificate? = null,
+    colorShade: PersonColorShade,
+    isCombinedCertificate: Boolean,
+    statusBadgeText: Int = 0,
+    badgeCount: Int = 0,
+    onCovPassInfoAction: () -> Unit
+) {
+    val valid = primaryCertificate.isValid
+    val context = root.context
+    val color = when {
+        primaryCertificate.isValid -> colorShade
+        else -> PersonColorShade.COLOR_INVALID
+    }
+    backgroundImage.setImageResource(color.background)
+    starsImage.setImageDrawable(starsDrawable(context, color))
+    name.text = primaryCertificate.fullName
+
+    qrCodeCard.apply {
+        image.loadAny(primaryCertificate.getValidQrCode(Locale.getDefault().language)) {
+            crossfade(true)
+            loadingView(qrCodeCard.image, qrCodeCard.progressBar)
+        }
+        statusText.isVisible = valid
+        statusBadge.isVisible = valid
+        statusBadge.text = context.getString(statusBadgeText)
+        covpassInfoTitle.isVisible = valid
+        covpassInfoButton.isVisible = valid
+        covpassInfoButton.setOnClickListener { onCovPassInfoAction() }
+        invalidOverlay.isGone = valid
+        image.isEnabled = valid
+        certificateToggleGroup.isVisible = isCombinedCertificate && valid
```

Comment:
The admission State `Other` is the only one that can contain invalid certificates (e.g. an outdated RAT test). Therefore, we don't need the `&& valid` here.

Reply:
Yep, you're right. I think  i can ditch the `isCombinedCertificate` boolean alltogether and just check if `secondaryCertificate` is not null instead.